src/smc-webapp/project_page.cjsx

```diff
@@ -247,6 +247,52 @@ FreeProjectWarning = rclass ({name}) ->
             {@extra(host, internet)}
         </Alert>
 
+DiskSpaceWarning = rclass ({name}) ->
+    displayName : 'DiskSpaceWarning'
+
+    reduxProps :
+        projects :
+            project_map              : rtypes.immutable.Map
+            get_total_project_quotas : rtypes.func
+
+    propTypes :
+        project_id : rtypes.string
+
+    shouldComponentUpdate: (nextProps) ->
+        return @props.project_map?.get(@props.project_id) != nextProps.project_map?.get(nextProps.project_id)
+
+    render: ->
+        if not require('./customize').commercial
+            return null
+        if @props.free_warning_closed
+            return null
+        quotas = @props.get_total_project_quotas(@props.project_id)
+        project_status = @props.project_map?.get(@props.project_id)?.get('status')
+        if not quotas?.disk_quota? or not project_status?
+            return null
+        else
+            disk = project_status.get('disk_MB') ? 0
+            if disk?
```

Comment:
This line makes absolutely no sense at all, since disk is DEFINITELY defined, based on the previous line.